---
title: 'R language basics, part 3: factor'
createTime: 2025/12/03 23:46:41
permalink: /docs/r-course/talk04/
---
## R language basics, part 3: factor

::: note
Reference: [R for Data Science](https://r4ds.had.co.nz)

The book updated to 2^nd^ ed. on July,2023, here’ s a [link](https://r4ds.hadley.nz) to the official website.

  ::: flex around center
  <LinkCard 
    icon="ic:outline-description"
    title="View the original slide"
    href="[/docs/hust-gp-template/faqs/](https://github.com/Lucas04-nhr/R-for-Data-Science/blob/main/talk04.pdf)" > 
  </LinkCard>

  <LinkCard 
    icon="ic:round-document-scanner"
    title="View the original Rmd file"
    href="https://github.com/Lucas04-nhr/R-for-Data-Science/blob/main/talk04.Rmd" > 
  </LinkCard>
  :::
  
:::

## IO and working enviroment management

Each R session is a separate **work space** containing its own data, variables, and operation history.

<img src="https://cdn.jsdelivr.net/gh/Lucas04-nhr/Pictures@main/uPic/r_sessions.png" alt="r_sessions" />

Each RStudio session is automatically associated with a R session

> Not only RStudio, PyCharm or VSCode also support R session.
>
> However, I’m keen on coding with PyCharm but not RStudio, for its wonderful Plug-in Environment, which can let me use plug-ins such as Code GeeX by Zhipu AI (a company founded by some student in KEG team inTsinghua University) or GitHub Copilot by GitHub to let the coding process more quickly, for the instruction from GPTs.

<img src="https://cdn.jsdelivr.net/gh/Lucas04-nhr/Pictures@main/uPic/r_session_in_rstudio.png" alt="r_session_in_rstudio"  />

If you want to coding with R using PyCharm or other JetBrains IDE (i.e. IntelliJ, CLion, etc.), remember to install the *R Language Plug-in*

<img src="https://cdn.jsdelivr.net/gh/Lucas04-nhr/Pictures@main/uPic/image-20231202111619635.png" alt="r-language plug-in in PyCharm" style="zoom:50%;" />

For instruction how to get FREE Student Lisence of GitHub Pro, GitHub Copilot and JetBrains Products and their benefits, see their official website:

- [GitHub Global Campus](https://campus.github.com)

	Make sure you don’t use VPNs and use your phone to log in and apply (HUST Campus Network is recommended), give “Precise Location” permission to your browser. You may use your “[Student Number]@hust.edu.cn” mail to verify your identity as a student studying in HUST.

- [JetBrains Products](https://www.jetbrains.com/lp/leaflets-gdc/students/)

	Because our email addresses ending with “@hust.edu.cn” are banned due to misuse, you should apply for an online verification report on [CHSI](chsi.com.cn) (press the link to visit the website), instructions [here](https://www.chsi.com.cn/xlcx/rhsq.jsp).

### Start a new RStudio session by creating a new project

> To start a new session in PyCharm, simply press the bottom corner and select a new session.

- Click the Project button in the upper right corner and select New Project in the pop-up menu …

	<img src="https://cdn.jsdelivr.net/gh/Lucas04-nhr/Pictures@main/uPic/rstudio_create_new_project.png" alt="rstudio_create_new_project" style="zoom:48%;" />

- Select: New directory -> New Project in the popup window

	<img src="https://cdn.jsdelivr.net/gh/Lucas04-nhr/Pictures@main/uPic/rstudio_create_project_2.png" alt="rstudio_create_project_2" style="zoom:48%;" />

- Enter a new directory name, choose its mother directory ...

	<img src="https://cdn.jsdelivr.net/gh/Lucas04-nhr/Pictures@main/uPic/rstudio_create_new_project_3.png" alt="rstudio_create_new_project_3" style="zoom:50%;" />

### Working Space

Current workspace, including all loaded data, packets and homebrew functions.

Variables can be managed with the following code:

```R
ls() # Show all the variables in current workspace/session
rm(x) # Remove a variable
rm(list = ls()) # Remove ALL variables in current workspace/session
```

### Variables in working space in RStudio

The "Environment" window in the upper right corner of RStudio shows all the variables of the current workspace.

<img src="https://cdn.jsdelivr.net/gh/Lucas04-nhr/Pictures@main/uPic/rstudio_enviroment_window.png" alt="rstudio_enviroment_window" style="zoom:50%;" />

### Save and restore work space

```R
# Save all loaded variables into an external .RData file
save.image(file = "prj_r_for_bioinformatics_aug3_2019.RData") 
# Restore (load) saved work space
load(file = "prj_r_for_bioinformatics_aug3_2019.RData")
```

**Notes:**

- Existing variables will be kept, however, those will the same names will be replaced by loaded variables

- Please consider using `rm(list=ls())` to remove all existing variables to have a clean start

- You may need to reload all the packages

### Save selected variables

Sometimes you need to transfer processed data to a collaborator ...

```R
# Save selected variables to external
save(
  city, 
  country, 
  file="1.RData"
)
# You can specify directory name
load("1.RData")
```

### Close and (re)open a project

- To close a project

	<img src="https://cdn.jsdelivr.net/gh/Lucas04-nhr/Pictures@main/uPic/Rstudio_close_a_project.png" alt="Rstudio_close_a_project" />

- In RStudio and similar IDEs, there are some preferences to choose

	<img src="https://cdn.jsdelivr.net/gh/Lucas04-nhr/Pictures@main/uPic/rstudio_project_options.png" alt="rstudio_project_options" style="zoom:80%;" />

> The UI in PyCharm

<img src="https://cdn.jsdelivr.net/gh/Lucas04-nhr/Pictures@main/uPic/pycharm_project_options.png" alt="pycharm_project_options" style="zoom: 33%;" />

**Notes:**

- Save on exit
- Load on opening 
- When the data is large, the loading time may be too long ...

### Open a project

<img src="https://cdn.jsdelivr.net/gh/Lucas04-nhr/Pictures@main/uPic/rstudio_open_a_project.png" alt="rstudio_open_a_project" style="zoom:80%;" />

> When in PyCharm, simply drag the working directory to its main window, remember to trust the project.
>

## Factors

Factor is a data structure used for fields that takes only predefined, finite number of values (categorical data).

It will limit the selection of input data.

### Play around with `levels()`

Here are instructions of modifying factor levels

> Based on the textbook

The levels are terse and inconsistent. Let’s tweak them to be longer and use a parallel construction. Like most rename and recoding functions in the tidyverse, the new values go on the left and the old values go on the right:

```R
load(gss_cat)

mutate(
  partyid = fct_recode(partyid,
    "Republican, strong"    = "Strong republican",
    "Republican, weak"      = "Not str republican",
    "Independent, near rep" = "Ind,near rep",
    "Independent, near dem" = "Ind,near dem",
    "Democrat, weak"        = "Not str democrat",
    "Democrat, strong"      = "Strong democrat"
  )
) 

count(partyid)

#> # A tibble: 10 × 2
#>   partyid                   n
#>   <fct>                 <int>
#> 1 No answer               154
#> 2 Don't know                1
#> 3 Other party             393
#> 4 Republican, strong     2314
#> 5 Republican, weak       3032
#> 6 Independent, near rep  1791
#> # ℹ 4 more rows
```

Use this technique with care: if you group together categories that are truly different you will end up with misleading results.

The order of the `levels` determines the sorting order.

### Use factor to clean data

> Usage of `fct_xxx()` functions.

Suppose I have a set of gender data that is written in a very irregular way:

```R
gender =
	c("f", "m ", "male ","male", "female", "FEMALE", "Male", "f", "m")

gender_fct =
  as.factor(gender)

fct_count(gender_fct)
```

The output looks like this:

<img src="https://cdn.jsdelivr.net/gh/Lucas04-nhr/Pictures@main/uPic/image-20231202114121080.png" alt="image-20231202114121080" style="zoom:50%;" />

Now I request to replace with Female, Male.

```R
gender_fct =
  fct_collapse(
    gender,
    Female = c("f", "female", "FEMALE"),
    Male = c("m ", "m", "male ", "male", "Male")
  )

fct_count(gender_fct)
```

<img src="https://cdn.jsdelivr.net/gh/Lucas04-nhr/Pictures@main/uPic/image-20231202114937281.png" alt="image-20231202114937281" style="zoom:50%;" />

You can also use `fct_relabel()` to do the same thing

```R
fct_relabel(
  gender,
  ~ ifelse(
    tolower(
      substring(., 1, 1)) == "f",
    "Female",
    "Male"
  )
)
```

### Usage of factors in drawing plots

```R
library(ggplot2)

responses =
  factor(
    c("Agree", "Agree", "Strongly Agree", "Disagree", "Disagree", "Agree")
  )

response_barplot =
  ggplot(
    data = data.frame(responses),
    aes(x = responses)
  ) +
  geom_bar()
```

<img src="https://cdn.jsdelivr.net/gh/Lucas04-nhr/Pictures@main/uPic/response_barplot.png" alt="response_barplot" style="zoom:24%;" />

By default, `factor` is sorted alphabetically.

 `ggplot2` also plots `factor` in that order, so you can adjust the `factor` to adjust the drawing order.

```R
res =
  data.frame(responses)
# Sort by level of agreement from strong -> weak
res$res =
  factor(
    res$res,
    levels =
       c("Strongly Agree", "Agree", "Disagree")
  )

response_barplot2 =
  ggplot(
    data = res,
    aes(x = res)
  ) +
  geom_bar() +
  xlab("Response")
```

<img src="https://cdn.jsdelivr.net/gh/Lucas04-nhr/Pictures@main/uPic/response_barplot2.png" alt="response_barplot2" style="zoom:24%;" />

You can also use the parameter `ordered` to let others know that your `factor` is ordered properly.

```R
responses =
  factor(
    c("Agree", "Agree", "Strongly Agree", "Disagree", "Disagree", "Agree"),
    ordered = TRUE
  )
```

<img src="https://cdn.jsdelivr.net/gh/Lucas04-nhr/Pictures@main/uPic/image-20231202134409231.png" alt="image-20231202134409231" style="zoom:50%;" />

### Using `factor` to vhange values

You can use `recode()` in `dplyr` package to change `value`

`dplyr` is a grammar of data manipulation, providing a consistent set of verbs that help you solve the most common data manipulation challenges:

+ `mutate()` adds new variables that are functions of existing variables
+ `select()` picks variables based on their names.
+ `filter()` picks cases based on their values.
+ `summarise()` reduces multiple values down to a single summary.
+ `arrange()` changes the ordering of the rows.

These all combine naturally with `group_by()` which allows you to perform any operation “by group”. You can learn more about them in `vignette("dplyr")`. As well as these single-table verbs, dplyr also provides a variety of two-table verbs, which you can learn about in `vignette("two-table")`.

> Based on the introduction on the [official website](https://dplyr.tidyverse.org) of `dplyr`.

Here’s an example:

```R
x =
  factor(
    c("alpha", "beta", "gamma", "theta", "beta", "alpha")
  )

x =
  recode(
    x,
    alpha = "a",
    beta = "b",
    gamma = "c",
    theta = "d"
  )
```

<img src="https://cdn.jsdelivr.net/gh/Lucas04-nhr/Pictures@main/uPic/image-20231202135555790.png" alt="image-20231202135555790" style="zoom:50%;" />

### Delete useless `levels`

```R
mouse.genes =
  read.delim(
    file = "data/talk04/mouse_genes_biomart_sep2018.txt",
    sep = "\t",
    header = T, 
    stringsAsFactors = T 
  )
```

<img src="https://cdn.jsdelivr.net/gh/Lucas04-nhr/Pictures@main/uPic/image-20231202135803901.png" alt="image-20231202135803901" style="zoom:50%;" />

If you draw a plot without deleting the useless `levels`, you will get this result:

<img src="https://cdn.jsdelivr.net/gh/Lucas04-nhr/Pictures@main/uPic/mouse_gene_plot01.png" alt="mouse_gene_plot01.png" style="zoom:50%;" />

But when you delete the useless `level` using these commands:

```R
mouse.chr_10_12$Chromosome.scaffold.name =
  droplevels(mouse.chr_10_12$Chromosome.scaffold.name)
```

You will see that:

<img src="https://cdn.jsdelivr.net/gh/Lucas04-nhr/Pictures@main/uPic/image-20231202141212937.png" alt="image-20231202141212937" style="zoom:50%;" />

Then, you’ll get the plot like this:

<img src="https://cdn.jsdelivr.net/gh/Lucas04-nhr/Pictures@main/uPic/mouse_gene_plot02.png" alt="mouse_gene_plot02" style="zoom:24%;" />

*Source code:*

``` R
mouse_gene_plot02 =
  ggplot(
    mouse.chr_10_12,
    aes(
      x = Chromosome.scaffold.name,
      y = Transcript.length..including.UTRs.and.CDS.
    )
  ) +
  geom_boxplot() +
  labs(
    x = "Chromosome Scaffold Name",
    y = "Transcript Length (including UTRs and CDS)"
  )
```

You can also use `tibble` to solve these problems:

```R
mouse.tibble =
  read_delim(
    file = "data/talk04/mouse_genes_biomart_sep2018.txt",
    delim = "\t",
    quote = "",
    show_col_types = FALSE
  )

mouse.tibble.chr10_12 =
  mouse.tibble %>% filter(
    `Chromosome/scaffold name` %in% c("10", "11", "12"))


mouse_gene_plot03 =
    ggplot(
        mouse.tibble.chr10_12,
        aes(
        x = Chromosome.scaffold.name,
        y = Transcript.length..including.UTRs.and.CDS.
        )
    ) +
    geom_boxplot() +
    labs(
        x = "Chromosome",
        y = "Transcript length (bp)"
    ) +
    coord_flip() +
    ylim(0, 2500) +
    theme_bw()
```

<img src="https://cdn.jsdelivr.net/gh/Lucas04-nhr/Pictures@main/uPic/mouse_gene_plot03.png" alt="mouse_gene_plot03" style="zoom:24%;" />

### Advance usage

- Use `reorder()` function to reorder the level.

	```R
	x = reorder( 
	  `Chromosome/scaffold name`,
	  `Transcript length (including UTRs and CDS)`,
	  median
	)
	```

- Use `forcats::fct_reorder()` to reorder factors

	```R
	x = fct_reorder( 
	  `Chromosome/scaffold name`,
	  `Transcript length (including UTRs and CDS)`,
	  median 
	)
	```
