---
title: Molecular Biology Homework 05
createTime: 2024/10/12
permalink: /blog/molecular-biology-homework-05/
excerpt: "Archived molecular biology homework synthesizing comprehensive knowledge on molecular mechanisms and biological applications."
tags:
  - Molecular Biology
  - Homework
  - Archived
---

::: warning

All the posts with "Archived" tag are old notes for backup only. They are migrated from my old blog system with `Jekyll`, with lack of proper formatting.

所有带有 “Archived” 标签的文章均为旧笔记，仅供备份存档。它们是从我以前使用 `Jekyll` 博客系统迁移过来的，缺乏更新的格式支持。

:::

## Topic of Contents & Thoughts

> Some are based on the textbook and Wikipedia

1. Centroparticles contribute to the tissue and separation of chromosomes.

2. Some general rules in replication：

- **bi-directional**
- **semi-conservative**
- **semi-discontinuous**

3. **Okazaki Fragments** are short sequences of DNA nucleotides (approximately 150 to 200 base pairs long in eukaryotes) which are synthesized discontinuously and later linked together by the enzyme DNA ligase to create the lagging strand during DNA replication.They were discovered in the 1960s by the Japanese molecular biologists Reiji and Tsuneko Okazaki, along with the help of some of their colleagues.[^1]

   > During DNA replication, the double helix is unwound and the complementary strands are separated by the enzyme DNA helicase, creating what is known as the DNA replication fork. Following this fork, DNA primase and DNA polymerase begin to act in order to create a new complementary strand. Because these enzymes can only work in the $5^\prime$ to $3^\prime$ direction, the two unwound template strands are replicated in different ways.One strand, the leading strand, undergoes a continuous replication process since its template strand has $3^\prime$ to $5^\prime$ directionality, allowing the polymerase assembling the leading strand to follow the replication fork without interruption. The lagging strand, however, cannot be created in a continuous fashion because its template strand has $5^\prime$ to $3^\prime$ directionality, which means the polymerase must work backwards from the replication fork. This causes periodic breaks in the process of creating the lagging strand. The primase and polymerase move in the opposite direction of the fork, so the enzymes must repeatedly stop and start again while the DNA helicase breaks the strands apart. Once the fragments are made, DNA ligase connects them into a single, continuous strand.The entire replication process is considered "semi-discontinuous" since one of the new strands is formed continuously and the other is not.

4. The replication fork is a structure that forms within the long helical DNA during DNA replication. It is produced by enzymes called helicases that break the hydrogen bonds that hold the DNA strands together in a helix. The resulting structure has two branching "prongs", each one made up of a single strand of DNA. These two strands serve as the template for the leading and lagging strands, which will be created as DNA polymerase matches complementary nucleotides to the templates; the templates may be properly referred to as the leading strand template and the lagging strand template.[^2]

   > **DNA is read by DNA polymerase in the 3′ to 5′ direction, meaning the new strand is synthesized in the 5' to 3' direction.** Since the leading and lagging strand templates are oriented in opposite directions at the replication fork, a major issue is how to achieve synthesis of new lagging strand DNA, whose direction of synthesis is opposite to the direction of the growing replication fork.

- **Leading strand** is the strand of new DNA which is synthesized in the same direction as the growing replication fork. This sort of DNA replication is continuous.
- **Lagging strand** is the strand of new DNA whose direction of synthesis is opposite to the direction of the growing replication fork. Because of its orientation, replication of the lagging strand is more complicated as compared to that of the leading strand. As a consequence, the DNA polymerase on this strand is seen to "lag behind" the other strand.

5. **DNA polymerase** is a member of a family of enzymes that catalyze the synthesis of DNA molecules from nucleoside triphosphates, the molecular precursors of DNA. These enzymes are essential for DNA replication and usually work in groups to create two identical DNA duplexes from a single original DNA duplex. During this process, DNA polymerase "reads" the existing DNA strands to create two new strands that match the existing ones.[^3]

   > $\text{deoxynucleoside triphosphate} + DNA_n \Leftrightarrow  \text{pyrophosphate} + DNA_{n+1}$
   >
   > The known DNA polymerases have highly conserved structure, which means that their overall catalytic subunits vary very little from species to species, independent of their domain structures. Conserved structures usually indicate important, irreplaceable functions of the cell, the maintenance of which provides evolutionary advantages. The shape can be described as resembling a right hand with thumb, finger, and palm domains. The palm domain appears to function in catalyzing the transfer of phosphoryl groups in the phosphoryl transfer reaction. DNA is bound to the palm when the enzyme is active. This reaction is believed to be catalyzed by a two-metal-ion mechanism. The finger domain functions to bind the nucleoside triphosphates with the template base. The thumb domain plays a potential role n the processivity, translocation, and positioning of the DNA.

6. The "wrong" mutual variant of the base group will form a mismatch, which will be removed from the polymerase active site to the exoctase active site and be removed.

7. Some DNA polymerases have a high error rate or low continuous operation ability, mainly for DNA repair. Because there are usually few error sites, even with a high error rate, there is little error, and there is no need for high continuous operation.

8. Some low-fidelity DNA polymerases can continue to synthesize new chains opposite to the error fragment in the template.

9. **Primer-template junction**[^4]

   > DNA polymerase's rapid catalysis due to its processive nature. Processivity is a characteristic of enzymes that function on polymeric substrates. In the case of DNA polymerase, the degree of processivity refers to the average number of nucleotides added each time the enzyme binds a template. The average DNA polymerase requires about one second locating and binding a primer/template junction. Once it is bound, a nonprocessive DNA polymerase adds nucleotides at a rate of one nucleotide per second. 207–208  Processive DNA polymerases, however, add multiple nucleotides per second, drastically increasing the rate of DNA synthesis. The degree of processivity is directly proportional to the rate of DNA synthesis. The rate of DNA synthesis in a living cell was first determined as the rate of phage T4 DNA elongation in phage infected E. coli. During the period of exponential DNA increase at 37 °C, the rate was 749 nucleotides per second.

## Volcabular

|       **English**        |
| :----------------------: |
| primer:template junction |
|     DNA polymerases      |
|  degree of processivity  |
| proofreading exonuclease |
|       exonuclease        |
|       endonuclease       |
|     replication fork     |
|      leading strand      |
|      lagging strand      |
|    Okazaki fragments     |
|         primase          |
|       DNA helicase       |
|         RNase H          |
|        DNA ligase        |
|       DNA helicase       |
|         polarity         |
|  ssDNA-binding protein   |
|   cooperative binding    |
|       topoisomers        |
|         telomere         |

## Translation

> The forth slide corresponds to Chapter 9 and 10 in the textbook.

### Chapter 9

#### Original Text

DNA synthesis is dependent on the presence of two types of substrates: the four deoxynucleoside triphosphates (dATP, dGTP, dCTP, and dGTP) and the template DNA structure, a primer:template junction. The template DNA determines the sequence of incorporated nucleotides. The primer serves as the substrate for deoxynucleotide addition, each being added successively to the OH at its 30 end.

DNA synthesis is catalyzed by an enzyme called DNA polymerase that uses a single active site to add any of the four dNTP precursors. Structural studies of DNA polymerases reveal that these enzymes resemble a hand that grips the DNA and incoming nucleotide in the catalytic site. DNA polymerases are processive: Each time they bind a substrate, they add many nucleotides. Proofreading exonucleases further enhance the accuracy of DNA synthesis by acting like a “delete key” that removes incorrectly added nucleotides.

In the cell, both strands of a DNA template are duplicated simultaneously at a structure called the replication fork. Because the two strands of the DNA are antiparallel, only one of the template DNA strands can be replicated in a continuous fashion (called the leading strand). The other DNA strand (called the lagging strand) must be synthesized first as a series of short DNA fragments, called Okazaki fragments. Each DNA strand is initiated with an RNA primer that is synthesized by an enzyme called primase. These primers must be removed to complete the replication process. After the replacement of the RNA primers with DNA, all of the separately primed lagging-strand DNA fragments are joined together to form one continuous DNA strand by DNA ligase.

An array of proteins in addition to the DNA polymerases coordinates and facilitates the DNA replication reaction. These additional factors facilitate the unwinding of the dsDNA template (DNA helicase), stabilize the ssDNA template (SSBs), and remove supercoils generated in front of the replication fork (topoisomerases). DNA polymerases are specialized to perform different events during DNA replication. Some are designed to be highly processive and others, only weakly processive. DNA sliding clamps enhance the processivity of the DNA polymerases that replicate large regions of DNA. These clamp proteins are topologically linked to DNA, but they are able to slide along the recently synthesized DNA while bound to the DNA polymerase. This interaction effectively prevents the attached DNA polymerase from dissociating from the primer:template junction. Special protein complexes called sliding DNA clamp loaders use the energy of ATP binding and hydrolysis to place sliding clamps on the DNA near primer:template junctions.

Interactions between the proteins at the replication fork have an important role in DNA synthesis. In _E. coli_, the three DNA polymerases are part of a large complex called the DNA Pol III holoenzyme. Binding of the DNA Pol III holoenzyme to the DNA helicase stimulates the rate of DNA unwinding. Similarly, binding of primase to the DNA helicase increases its ability to synthesize RNA primers. Thus, the replication reaction works best when the entire array of replication proteins is present at the replication fork. Together, this set of proteins forms a complex called the replisome.

The initiation of DNA replication is directed by specific DNA sequences called replicators. The physical site of replication initiation is called an origin of replication. The replicator is specifically bound bya protein called the initiator, which stimulates the recruitment of other proteins required for the initiation of replication (such as DNA helicase) and, in some but not all cases, the unwinding of the origin DNA. The subsequent events in the initiation of DNA replication are largely driven by either protein – protein or nonspecific protein – DNA interactions.

In eukaryotic cells, the initiation of DNA replication is tightly regulated to ensure that every nucleotide of every chromosome is replicated once and only once per round of cell division. This tight regulation is accomplished by controlling loading and activation of the replicative helicase during the cell cycle. During the $G_1$ phase of the cell cycle, helicases can be loaded but not activated. During the remainder of the cell cycle (the $S$, $G_2$, and $M$ phases), loaded helicases can be activated, leading to the initiation of DNA replication, but no new helicase loading can occur. Thus, each replicator can direct only one round of replication initiation per cell cycle, ensuring that the DNA is replicated exactly once.

Finishing DNA replication requires the action of specific enzymes. For circular chromosomes, type II DNA topoisomerases separate the topologically linked circular products from one another. Linear chromosomes also require special proteins to ensure their complete replication. In eukaryotic cells, a specialized DNA polymerase called telomerase allows the ends of the chromosome (called telomeres) to act as a unique origin of replication. By extending the $3^\prime$ ends of the telomere, telomerase eliminates the progressive loss of chromosome ends that conventional DNA synthesis by the replication fork machinery would cause. Proteins bound to telomeric DNA act to regulate the activity of telomerase and protect the ends of chromosomes from degradation and recombination.

#### Translated Text

DNA 合成依赖于两种类型底物的存在：四种三磷酸脱氧核苷酸（dATP、dGTP、dCTP 和 dGTP）和模板 DNA 结构，即引物：模板结合点。模板 DNA 决定了合并核苷酸的序列。引物作为脱氧核苷酸添加的底物，每个引物都被连续地添加到其 3' 端的羟基上。

DNA 合成由一种名为 DNA 聚合酶的酶催化，它使用单个活性位点添加四种 dNTP 前体中的任何一种。DNA 聚合酶的结构研究显示，这些酶类似于握住 DNA 和进入催化位点的核苷酸的手。DNA 聚合酶是过程性的：每次它们结合一个底物时，它们会添加许多核苷酸。校对外切酶通过像“删除键”一样移除错误添加的核苷酸，进一步提高了 DNA 合成的准确性。

在细胞中，DNA 模板的两条链同时在一个称为复制叉的结构处复制。由于 DNA 的两条链是反向平行的，只有模板 DNA 的一条链可以连续地复制（称为主链）。另一条 DNA 链（称为滞后链）必须首先作为一系列短的 DNA 片段（称为冈崎片段）合成。每条 DNA 链都以由一种名为引物酶合成的 RNA 引物开始。这些引物必须被移除才能完成复制过程。在用 DNA 替换 RNA 引物后，所有单独引导的滞后链 DNA 片段将被 DNA 连接酶连接在一起，形成一条连续的 DNA 链。

除了 DNA 聚合酶外，一系列蛋白质协调并促进 DNA 复制反应。这些额外因子促进双链 DNA 模板的解旋（DNA 解旋酶），稳定单链 DNA 模板（SSBs），并去除在复制叉前方产生的超螺旋（拓扑异构酶）。DNA 聚合酶专门用于在 DNA 复制过程中执行不同事件。有些被设计为高度过程性，而另一些则只具有较弱的过程性。DNA 滑动夹增强了复制大片段 DNA 的 DNA 聚合酶的过程性。这些夹蛋白与 DNA 拓扑地连接在一起，但它们能够在与 DNA 聚合酶结合时沿着最近合成的 DNA 滑动。这种相互作用有效地防止附着的 DNA 聚合酶从引物:模板结合点解离。名为滑动 DNA 夹加载酶的特殊蛋白质复合物利用 ATP 结合和水解的能量将滑动夹放置在靠近引物:模板结合点的 DNA 上。

在复制叉处蛋白质之间的相互作用在 DNA 合成中起着重要作用。在大肠杆菌中，三种 DNA 聚合酶是 DNA Pol III 全酶复合物的一部分。DNA Pol III 全酶复合物与 DNA 解旋酶的结合刺激了 DNA 解旋速率。同样，引物酶与 DNA 解旋酶的结合增强了其合成 RNA 引物的能力。因此，当整个复制蛋白质组合出现在复制叉处时，复制反应效果最佳。这些蛋白质一起形成一个称为复制体的复杂体。

DNA 复制的起始由称为复制起始子的特定 DNA 序列指导。复制起始的物理位置称为复制起点。复制起始子特异地结合一种称为启动子的蛋白质，它刺激其他启动复制所需的蛋白质（如 DNA 解旋酶）的信号，并在某些情况下，解开起点 DNA。DNA 复制起始后续事件主要由蛋白质-蛋白质或非特异性蛋白质-DNA 相互作用驱动。

在真核细胞中，DNA 复制的起始受到严格调控，以确保每个染色体的每个核苷酸在每一轮细胞分裂中只复制一次。通过在细胞周期中控制复制螺旋酶的加载和激活来实现这种严格调控。在细胞周期的 $G_1$ 阶段，螺旋酶可以被加载但不被激活。在细胞周期的其余时间（$S$、$G_2$ 和 $M$ 阶段），已加载的螺旋酶可以被激活，使得 DNA 复制启动，但这些时期不会发生新的螺旋酶加载。因此，每个复制起始子在每个细胞周期中只能指导一轮复制启动，确保 DNA 只被复制一次。

完成 DNA 复制需要特定酶的作用。对于环状染色体，II 类 DNA 拓扑异构酶将拓扑连接的环状产物相互分离。线性染色体也需要特殊蛋白质来确保它们的完全复制。在真核细胞中，一种名为端粒酶的专门 DNA 聚合酶使染色体末端（称为端粒）成为独特的复制起点。通过延伸端粒的 $3^\prime$ 端，端粒酶消除了常规 DNA 复制叉机构所导致的染色体末端逐渐丢失的问题。结合端粒 DNA 的蛋白质调节端粒酶的活性，并保护染色体末端免受降解和重组的影响。

### Chapter 10

#### Original Text

Organisms can survive only if their DNA is replicated faithfully and is protected from chemical and physical damage that would change its coding properties. The limits of accurate replication and repair of damage are revealed by the natural mutation rate. Thus, an average nucleotide is likely to be changed by mistake only about once every $10^9$ times it is replicated, although error rates for individual bases can vary over a 10,000-fold range. Much of the accuracy of replication is inherent in the way DNA polymerase copies a template. The initial selection of the correct base is guided by complementary pairing. Accuracy is increased by the proofreading activity of DNA polymerase. Finally, in mismatch repair, the newly synthesized DNA strand is scanned by an enzyme that initiates replacement of DNA containing incorrectly paired bases. Despite these safeguards, mistakes of all types occur: base substitutions, small and large additions and deletions, and gross rearrangements of DNA sequences.

Cells have a large repertoire of enzymes devoted to repairing DNA damage that would otherwise be lethal or would alter DNA so as to engender damaging mutations. Some enzymes directly reverse DNA damage, such as photolyases, which reverse pyrimidine dimer formation. A more versatile strategy is excision repair, in which a damaged segment is removed and replaced through new DNA synthesis for which the undamaged strand serves as a template. In base excision repair, DNA glycosylases and endonucleases remove only the damaged nucleotide, whereas in nucleotide excision repair, a short patch of single-stranded DNA containing the lesion is removed. In _E. coli_, excision repair is initiated by the UvrABC endonuclease, which creates a bubble over the site of the damage and cuts out a 12-nucleotide segment of the DNA strand that includes the lesion. Higher cells perform nucleotide excision repair in a similar manner, but a much larger number of proteins are involved, and the excised, single-stranded DNA is 24 – 32 nucleotides long.

The most hazardous kind of damage is a DNA break. Recombinational DSB repair is a pathway that mends breaks in which the sequence across the break is copied from a different but homologous duplex. If no template for repair synthesis is available, breaks in DNA are mended by NHEJ, which rejoins the ends but in an error-prone manner. If the cell needs to replicate damaged DNA, translesion synthesis allows the cell to tolerate the lesion. Translesion synthesis enables replication to continue across damage that blocks the progression of a replicating DNA polymerase. Translation synthesis is primarily mediated by a distinct and widespread family of DNA polymerases that are able to perform DNA synthesis in a manner that, although not always accurate, does not depend on base pairing.

Mutagenesis and its repair are of concern to us because they permanently affect the genes that organisms inherit and because cancer is often caused by mutations in somatic cells.

#### Translated Text

生物只有在其 DNA 被正确复制并受到保护，使它免受会改变其编码特性的化学和物理损害时才能存活。准确复制和修复损伤的极限由自然突变率揭示。因此，平均核苷酸可能仅在每复制约 $10^9$ 次时才会因错误而改变一次，尽管个别碱基的错误率可能在 10,000 倍范围内变化。复制的准确性很大程度上源于 DNA 聚合酶复制模板的方式。正确碱基的最初选择是由互补配对引导的。准确性通过 DNA 聚合酶的校对活动得以提高。随后，在错配修复中，新合成的 DNA 链被一种酶扫描，该酶启动替换含有错误配对碱基的 DNA。尽管有这些保障措施，各种类型的错误仍会发生：碱基替换、小型和大型插入和缺失，以及 DNA 序列的大规模重排。

细胞拥有大量酶来修复那些可能致命的或会改变 DNA 以至产生有害突变的 DNA 损伤。一些酶可以直接逆转 DNA 损伤，比如光化酶，可以逆转嘧啶二聚体的形成。一种适用性更广的策略是切除后修复，其中一个受损片段被移除，并通过新合成的 DNA 片段进行替换，未受损链作为模板。在碱基切除修复中，DNA 脱氧核糖酶和内切酶仅移除受损核苷酸，而在核苷酸切除修复中，包含损伤的一小片单链 DNA 都被移除。在大肠杆菌中，切除修复由 UvrABC 内切酶启动，它在损伤部位上形成一个小泡，并切除包括损伤的 DNA 链上的 12 个核苷酸片段。更高级的细胞以类似的方式执行核苷酸切除修复，但涉及的蛋白质数量更多，被切除的单链 DNA 长 24 到 32 个核苷酸。

最危险的损伤是 DNA 断裂。重组性 DSB 修复是一种修复断裂的途径，其中跨越断裂的序列从不同但同源的双链复制。如果没有可用于修复合成的模板，DNA 断裂将通过 NHEJ 修复，它以一种更加容易出错的方式重新连接末端。如果细胞需要复制受损的 DNA，跨损伤合成使细胞能够容忍损伤。跨损伤合成使复制能够继续跨越阻碍复制 DNA 聚合酶进展的损伤进行。跨损伤合成主要由一类独特且广泛的 DNA 聚合酶家族介导，它们能够进行 DNA 合成，虽然不总是准确，但这种途径不依赖于碱基配对。

我们应该持续关注 DNA 的突变和其修复，因为它们会永久影响生物所继承的基因，并且癌症往往是由体细胞突变引起的。

## References

[^1]: [Okazaki fragments](https://en.wikipedia.org/wiki/Okazaki_fragments) , Wikipedia

[^2]: [DNA Replication - Replication Fork](https://en.wikipedia.org/wiki/DNA_replication#Replication_fork) , Wikipedia

[^3]: [DNA Polimerase](https://en.wikipedia.org/wiki/DNA_polymerase) , Wikipedia

[^4]: [DNA Polimerase - Processivity](https://en.wikipedia.org/wiki/DNA_polymerase#Processivity) , Wikipedia
